crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -55,26 +56,43 @@ struct Context {
 
     /// Manages and builds the `AcirVar`s to which the converted SSA values refer.
     acir_context: AcirContext,
+
+    initialized_arrays: HashSet<BlockId>,
+}
+
+#[derive(Clone)]
+pub(crate) struct AcirArray {
+    block_id: BlockId,
+    len: usize,
+}
+impl Debug for AcirArray {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "id: {}, len: {}", self.block_id.0, self.len)
+    }
 }
 
 #[derive(Debug, Clone)]
 pub(crate) enum AcirValue {
     Var(AcirVar, AcirType),
     Array(im::Vector<AcirValue>),
+    DynamicArray(AcirArray),
 }
 
 impl AcirValue {
     fn into_var(self) -> AcirVar {
         match self {
             AcirValue::Var(var, _) => var,
-            AcirValue::Array(_) => panic!("Called AcirValue::into_var on an array"),
+            AcirValue::DynamicArray(_) | AcirValue::Array(_) => {
+                panic!("Called AcirValue::into_var on an array")
+            }
         }
     }
 
     fn flatten(self) -> Vec<(AcirVar, AcirType)> {
         match self {
             AcirValue::Var(var, typ) => vec![(var, typ)],
             AcirValue::Array(array) => array.into_iter().flat_map(AcirValue::flatten).collect(),
+            AcirValue::DynamicArray(_) => unimplemented!("Cannot flatten a dynamic array"),
```

Comment:
why?

Reply:
Flattening is for nested arrays, which this PR does not support, and this is why I used 'unimplemented' here.